### ISSUE TITLE
custom dataset training

### ISSUE THREAD

**Issue by bipungithub** [google-research/google-research#758]
Hi there, 
I am able to run the coltran using custom_colorization.py on the set of images successfully. 
Could you give some direction on how to train for a custom dataset instead of imagenet?

**Comment by MechCoder** [comment]
I added support for training and finetuneing on custom datasets. You can set the FLAGS `dataset=custom` and `data_dir` to be a directory containing RGB images. 

See: https://github.com/google-research/google-research/commit/db430ca3f9575e05245898880a95889e8025a92e

If you are using the same resolutions as in our paper, finetuneing is recommended. You can set `pretrain_dir` to be a pretrained ImageNet (colorizer/upsampler) checkpoint.

**Comment by saaunmanpati** [comment]
> I added support for training and finetuneing on custom datasets. You can set the FLAGS `dataset=custom` and `data_dir` to be a directory containing RGB images.
> 
> See: [db430ca](https://github.com/google-research/google-research/commit/db430ca3f9575e05245898880a95889e8025a92e)
> 
> If you are using the same resolutions as in our paper, finetuneing is recommended. You can set `pretrain_dir` to be a pretrained ImageNet (colorizer/upsampler) checkpoint.

Thank you

**Comment by ketan-lambat** [comment]
- ### How long do I need to train on a custom dataset? 
- ### How do I know if the training is complete?
- ### How to get the result/output for a custom dataset?

I am using this Notebook, for training on the cityscapes dataset.
[Link to Colab Notebook](https://colab.research.google.com/drive/1hG5t2djdrToWGyQdvGJ3Sf_DkDfWi5jE?usp=sharing)

I trained the model on colab (till GPU resource got exhausted), got around 23 checkpoints.
```
I1115 14:34:04.833563 140478266419072 run.py:286] Saved checkpoint to /content/drive/MyDrive/Colab_Work/HONORS/ColTran-v2/google-research/coltran/logs/cityscapes_ckpt/model-20
I1115 14:37:47.372999 140478266419072 run.py:282] Loss: 0.549 bits/dim, Speed: 0.449 steps/second
I1115 14:41:29.732624 140478266419072 run.py:282] Loss: 0.507 bits/dim, Speed: 0.450 steps/second
I1115 14:45:12.382166 140478266419072 run.py:282] Loss: 0.507 bits/dim, Speed: 0.449 steps/second
I1115 14:48:54.532032 140478266419072 run.py:282] Loss: 0.497 bits/dim, Speed: 0.450 steps/second
I1115 14:52:36.791089 140478266419072 run.py:282] Loss: 0.529 bits/dim, Speed: 0.450 steps/second
I1115 14:52:40.168719 140478266419072 run.py:286] Saved checkpoint to /content/drive/MyDrive/Colab_Work/HONORS/ColTran-v2/google-research/coltran/logs/cityscapes_ckpt/model-21
I1115 14:56:22.615087 140478266419072 run.py:282] Loss: 0.530 bits/dim, Speed: 0.450 steps/second
I1115 15:00:05.021047 140478266419072 run.py:282] Loss: 0.522 bits/dim, Speed: 0.450 steps/second
I1115 15:03:47.351305 140478266419072 run.py:282] Loss: 0.489 bits/dim, Speed: 0.450 steps/second
I1115 15:07:29.647610 140478266419072 run.py:282] Loss: 0.528 bits/dim, Speed: 0.450 steps/second
I1115 15:11:11.850597 140478266419072 run.py:282] Loss: 0.550 bits/dim, Speed: 0.450 steps/second
I1115 15:11:15.348102 140478266419072 run.py:286] Saved checkpoint to /content/drive/MyDrive/Colab_Work/HONORS/ColTran-v2/google-research/coltran/logs/cityscapes_ckpt/model-22
I1115 15:14:58.002671 140478266419072 run.py:282] Loss: 0.542 bits/dim, Speed: 0.449 steps/second
I1115 15:18:40.301140 140478266419072 run.py:282] Loss: 0.501 bits/dim, Speed: 0.450 steps/second
I1115 15:22:22.593688 140478266419072 run.py:282] Loss: 0.528 bits/dim, Speed: 0.450 steps/second
I1115 15:26:05.144472 140478266419072 run.py:282] Loss: 0.499 bits/dim, Speed: 0.449 steps/second
```
So, till how long / which step / loss value should I continue to train?

Next, after training on a custom dataset, 
how to evaluate the model or obtain the results of colorized/recolorized images?

I used this cmd, but I guess it works only for imagenet dataset.

```
python -m coltran.run --config=coltran/configs/colorizer.py --mode=eval_valid --logdir=$LOGDIR --dataset=custom --data_dir=$EVAL_DATA_DIR 
```

Now, I am trying to use this (see the notebook for the next 2 steps)

```
python -m coltran.custom_colorize --config=coltran/configs/colorizer.py \
--logdir=$LOGDIR --img_dir=$IMG_DIR --store_dir=$STORE_DIR \
--mode=$MODE
```

Can someone please tell me if I am following the correct commands for getting the output?
A step-by-step guide would be appreciated. I am getting confused on which flow to follow.
